sdk-actors/src/test/java/io/dapr/actors/it/DaprIntegrationTestingRunner.java

```diff
@@ -0,0 +1,165 @@
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class DaprIntegrationTestingRunner {
+
+  private static AtomicInteger appGeneratorId = new AtomicInteger();
+
+  @Rule
+  public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+  public DaprIntegrationTestingRunner.DaprFreePorts DAPR_FREEPORTS;
+
+  private Runtime rt = Runtime.getRuntime();
+  private Process proc;
+
+  private String successMessage;
+  private Class serviceClass;
+  private Boolean useAppPort;
+  private Boolean useGrpcPort;
+  private Boolean useHttpPort;
+  private int sleepTime;
+  private String appName;
+  private boolean appRanOK = Boolean.FALSE;
+  private boolean isClient = false;
+
+  DaprIntegrationTestingRunner(String successMessage, Class serviceClass, Boolean useAppPort, Boolean useGrpcPort, Boolean useHttpPort, int sleepTime, boolean isClient) {
+    this.successMessage = successMessage;
+    this.serviceClass = serviceClass;
+    this.useAppPort = useAppPort;
+    this.useGrpcPort = useGrpcPort;
+    this.useHttpPort = useHttpPort;
+    this.sleepTime = sleepTime;
+    this.isClient = isClient;
+    this.generateAppName();
+    try {
+      DAPR_FREEPORTS = new DaprIntegrationTestingRunner.DaprFreePorts().initPorts();
+      environmentVariables.set("DAPR_HTTP_PORT", String.valueOf(DAPR_FREEPORTS.getHttpPort()));
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  public DaprFreePorts initializeDapr() throws Exception {
+    String daprCommand=this.buildDaprCommand();
+    System.out.println(daprCommand);
+    proc= rt.exec(daprCommand);
+
+    final Thread stuffToDo = new Thread(() -> {
+      try {
+        try (InputStream stdin = proc.getInputStream()) {
+          try(InputStreamReader isr = new InputStreamReader(stdin)) {
+            try (BufferedReader br = new BufferedReader(isr)){
+              String line;
+              while ((line = br.readLine()) != null) {
+                System.out.println(line);
+                if (line.contains(successMessage)) {
+                  this.appRanOK = true;
+                }
+              }
+            }
+          }
+
+        }
+      } catch (IOException ex) {
+        Assert.fail(ex.getMessage());
+      }
+    });
+    stuffToDo.start();
+    Thread.sleep(sleepTime);
+    return DAPR_FREEPORTS;
+  }
+
+  private static final String DAPR_RUN = "dapr run --app-id %s ";
+
+  // the arg in -Dexec.args is the app's port
+  private static final String DAPR_COMMAND = " -- mvn exec:java -Dexec.mainClass=%s -Dexec.classpathScope=test -Dexec.args=\"%s,%s\"";
```

Comment:
Add a comment that we're using "," because having multiple spaced tokens seems to fail on Linux.